lib/actions/form.js

```diff
@@ -128,3 +130,50 @@ export function formChanged (oldQuery, newQuery) {
     }
   }
 }
+
+/**
+ * Check if the trip should be replanned based on the auto plan strategy,
+ * whether the mobile view is active, and the old/new queries. Response type is
+ * an object containing various booleans.
+ */
+export function checkShouldReplanTrip (autoPlan, isMobile, oldQuery, newQuery) {
+  // Determine if either from/to location has changed
+  const fromChanged = !isEqual(oldQuery.from, newQuery.from)
+  const toChanged = !isEqual(oldQuery.to, newQuery.to)
+  const oneLocationChanged = (fromChanged && !toChanged) || (!fromChanged && toChanged)
+  // Check whether a trip should be auto-replanned
+  const strategy = isMobile && autoPlan?.mobile
+    ? autoPlan?.mobile
+    : autoPlan?.default
+  const shouldReplanTrip = evaluateAutoPlanStrategy(
+    strategy,
+    fromChanged,
+    toChanged,
+    oneLocationChanged
+  )
+  return {
+    fromChanged,
+    oneLocationChanged,
+    shouldReplanTrip,
+    toChanged
+  }
+}
+
+/**
+ * Shorthand method to evaluate auto plan strategy. It is assumed that this is
+ * being called within the context of the `formChanged` action, so presumably
+ * some query param has already changed. If further checking of query params is
+ * needed, additional strategies should be added.
+ */
+const evaluateAutoPlanStrategy = (strategy, fromChanged, toChanged, oneLocationChanged) => {
+  switch (strategy) {
+    case 'ONE_LOCATION_CHANGED':
+      if (oneLocationChanged) return true
+      break
+    case 'BOTH_LOCATIONS_CHANGED':
+      if (fromChanged && toChanged) return true
+      break
+    case 'ANY': return true
```

Comment:
What is this value for? Either remove it or document it in `example-config.yml`.

Reply:
https://github.com/opentripplanner/otp-react-redux/blob/8d500a2a4437843d3ebd6979469b1d7d49de28bd/example-config.yml#L21-L22